src/MassTransit.ApplicationInsights/ApplicationInsightsFilter.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright 2007-2018 Chris Patterson, Dru Sellers, Travis Smith, et. al.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+namespace MassTransit.ApplicationInsights
+{
+    using System;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using GreenPipes;
+    using MassTransit;
+    using Microsoft.ApplicationInsights;
+    using Microsoft.ApplicationInsights.DataContracts;
+
+
+    public class ApplicationInsightsFilter<T> : IFilter<T>
```

Comment:
Could this be a `ConsumeContext` filter instead of any filter? I can't imagine it being used in another context, or perhaps add a `where T : ConsumeContext` generic constraint?

Reply:
Could actually be `MessageContext` I suppose.

Reply:
Wished I understood how all the pipeline stuff worked, still trying to wrap my head around it.